static/js/components/CandidatesPreferences.jsx

```diff
@@ -0,0 +1,564 @@
+
+  const [selectedClassifications, setSelectedClassifications] = useState([]);
+  const [selectedAnnotationOrigin, setSelectedAnnotationOrigin] = useState();
+
+  const { handleSubmit, getValues, control, errors, reset } = useForm();
+
+  useEffect(() => {
+    reset({
+      groupIDs: Array(userAccessibleGroups.length).fill(false),
+    });
+    // Don't want to reset everytime the component rerenders and
+    // the defaultStartDate is updated, so ignore ESLint here
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [reset, userAccessibleGroups]);
+
+  // Set initial form values in the redux state
+  useEffect(() => {
+    dispatch(
+      candidatesActions.setFilterFormData({
+        savedStatus: "all",
+      })
+    );
+    // Don't want to reset everytime the component rerenders and
+    // the defaultStartDate is updated, so ignore ESLint here
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [dispatch]);
+
+  let formState = getValues({ nest: true });
+
+  const validateGroups = () => {
+    formState = getValues({ nest: true });
+    return formState.groupIDs.filter((value) => Boolean(value)).length >= 1;
+  };
+
+  const validateRedshifts = () => {
+    formState = getValues({ nest: true });
+    return (
+      // Both empty
+      (formState.redshiftMinimum === "" && formState.redshiftMaximum === "") ||
+      // Or both filled out
+      (formState.redshiftMinimum !== "" &&
+        formState.redshiftMaximum !== "" &&
+        parseFloat(formState.redshiftMaximum) >
+          parseFloat(formState.redshiftMinimum))
+    );
+  };
+
+  const validateSorting = () => {
+    formState = getValues({ nest: true });
+    return (
+      // All left empty
+      (formState.sortingOrigin === "" &&
+        formState.sortingKey === "" &&
+        formState.sortingOrder === "") ||
+      // Or all filled out
+      (formState.sortingOrigin !== "" &&
+        formState.sortingKey !== "" &&
+        formState.sortingOrder !== "")
+    );
+  };
+
+  const [addDialogOpen, setAddDialogOpen] = useState(false);
+
+  const onSubmit = async (formData) => {
+    const groupIDs = userAccessibleGroups.map((g) => g.id);
+    const selectedGroupIDs = groupIDs.filter(
+      (ID, idx) => formData.groupIDs[idx]
+    );
+    const data = {
+      id: Math.random().toString(36).substr(2, 5), // Assign a random ID to the profile
+      groupIDs: selectedGroupIDs,
+      savedStatus: formData.savedStatus,
+      default: true,
+    };
+    // decide if to show rejected candidates
+    if (formData.rejectedStatus) {
+      data.rejectedStatus = formData.rejectedStatus;
+    }
+    // Convert dates to ISO for parsing on back-end
+    if (formData.timeRange) {
+      data.timeRange = formData.timeRange;
+    }
+    if (formData.classifications.length > 0) {
+      data.classifications = formData.classifications;
+    }
+    if (formData.redshiftMinimum && formData.redshiftMaximum) {
```

Comment:
Related to above comment, perhaps handle these individually so someone can set a min or max but not necessarily both